### PR TITLE
ar/adding segment tracking to API

### PR DESCRIPTION
**Pull request by angyangie**
**What does this pull request do? Explain your changes. (required)**
https://github.com/livepeer/livepeerjs/issues/702
https://github.com/livepeer/livepeerjs/issues/706

**How did you test each of these updates (required)**
See here: https://app.segment.com/livepeer/sources/livepeer-com/debugger and here: https://app.segment.com/livepeer/sources/livepeer-com/schema/events for examples of new events being logged on Segment

Note: before pushing, need to add Segment API key to our `.env` on stage and prod.

**Does this pull request close any open issues?**
https://github.com/livepeer/livepeerjs/issues/702
https://github.com/livepeer/livepeerjs/issues/706

**Screenshots (optional):**

Example:

Screen Shot 2020-05-12 at 6 25 33 PM

**Checklist:**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by angyangie** [comment]
Confirmed with Mark that it is safe to remove HubSpot code and lose the HubSpot page view records going forward. They will live in Segment.

File: packages/api/src/controllers/helpers.js [hunk 1]
```diff
@@ -144,3 +145,27 @@ export async function sendgridEmail({
...
+  analytics.identify({
+    userId: userId,
+    traits: {
+      email: email,
+    },
+  })
+
+  const properties = {}
+  if ('properties' in event) {
+    for (const key in Object.keys(event.properties)) {
```
Comment (darkdarkdragon) [inline packages/api/src/controllers/helpers.js]: Here should be `for (const key of Object.keys(event.properties)) {`
`in` iterates over object properties, but Object.keys returns array.

File: packages/api/src/controllers/helpers.js [hunk 2]
```diff
@@ -144,3 +145,27 @@ export async function sendgridEmail({
...
   await SendgridMail.send(msg)
 }
+
+export async function trackAction(userId, email, event, segmentApiKey) {
+  var analytics = new SegmentAnalytics(segmentApiKey, { flushAt: 1 })
+  // note: for development, add `{ flushAt: 1 }` as a second parameter to SegmentAnalytics
+  analytics.identify({
+    userId: userId,
+    traits: {
+      email: email,
```
Comment (darkdarkdragon) [inline packages/api/src/controllers/helpers.js]: you could just write `traits: { email },`

**Review by darkdarkdragon — CHANGES_REQUESTED** [review]

**Review by darkdarkdragon — APPROVED** [review]
LGTM

**Pull request merged**